### PR TITLE
feat: Make v1alpha1 LogPipeline input disableable

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Make the application input disableable as a preparation for https://github.com/kyma-project/telemetry-manager/pull/1463. The input is enabled by default

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/pull/1463

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

File: internal/reconciler/logpipeline/reconciler_test.go [hunk 1]
```diff
@@ -89,8 +90,8 @@ func TestReconcile(t *testing.T) {
 		require.True(t, *updatedPipeline.Status.UnsupportedMode)
 	})
 
-	t.Run("should set status UnsupportedMode false if does not contains custom plugin", func(t *testing.T) {
```
Comment (shorim) [inline internal/reconciler/logpipeline/reconciler_test.go:92]: Is there a reason to remove this unit test?

**Comment by shorim** [comment]
Should we add an e2e test to check that no logs are collected when the application input is disabled?

File: internal/reconciler/logpipeline/reconciler_test.go (hunk 1 above)
Comment (skhalash) [reply internal/reconciler/logpipeline/reconciler_test.go:92]: You are right, it was not intended, will revert it

**Comment by skhalash** [comment]
Not sure, I thought a unit test would be enough. I could add it if you think it's necessary.

**Comment by skhalash** [comment]
@shorim Added a test

File: internal/reconciler/logpipeline/reconciler_test.go (hunk 1 above)
Comment (skhalash) [reply internal/reconciler/logpipeline/reconciler_test.go]: Done

**Review by nataliasitko — APPROVED** [review]

**Review by shorim — APPROVED** [review]

**Pull request merged**